### ISSUE TITLE
DragonFF Export - Wrong Right To Render Bytes for Shadow the Hedgehog

### ISSUE THREAD

**Issue by dreamsyntax** [Parik27/DragonFF#321]
Very minor issue I have noticed. DragonFF successfully handles most RW structs properly when dealing with models from Shadow the Hedgehog game. One thing noticed is the player model lighting is incorrect, due to a wrong byte set.

Using RWAnalyze after exporting a model with DragonFF:

Navigate to any Atomics with "Right To Render", and hex edit or replace them with the original file. As of this commit, usually only one byte is incorrect, which results in wrong material rendering. 

WRONG (DragonFF output):
```
16 01 00 00 01 00 00 00 
```
CORRECT (original):
```
16 01 00 00 02 00 00 00
```

![image](https://github.com/Parik27/DragonFF/assets/14857235/bf0ce910-05b4-4aca-a87a-ca6af19f226a)

All other RW structs are correct on export, which is a great achievement. Thank you for your work!

**Comment by Psycrow101** [comment]
@dreamsyntax is there any documentation about Right To Render so that this value can be changed in the GUI in a user-friendly way?

**Comment by Psycrow101** [comment]
In the latest update, the "Right To Render" value can be changed in the object properties
![r2r](https://github.com/user-attachments/assets/b7a86e05-c6fe-4208-b9f2-07ff60b3fb7a)

**Comment by dreamsyntax** [comment]
I have not identified any patterns nor have any knowledge about the struct.

While not ideal, I was hoping for user flow
1. Import an existing DFF
2. Edits/Changes are made
3. Export as DFF

The Right To Render bytes would be taken from the initial import.

Alternatively allowing for a donor right to render section to be supplied or a donor dff. Though in the case of donor DFF not sure how multiple RTR structs would be handled.


I don't think we know enough for a user friendly naming scheme. Maybe instead expose all the possible fields of the struct? Im not even sure the sizing is consistent so this may be a challenge.

**Comment by Psycrow101** [comment]
In the current version the last four bytes are taken from the import and can be changed with an integer value from 0

**Comment by MadGamerHD** [comment]
When exporting a vehicle model, what should I set the 'Pipeline' to? And what value should I use for 'Right to Render' on the object?
I'm not exactly sure what the Pipeline is for, or how it affects the export process.

Also, after exporting, I checked the .dff file but couldn't find the RW Right to Render flag—it's either missing or not set correctly. How do I make sure it's included during export?

![Image](https://github.com/user-attachments/assets/df071c0b-832a-41f0-b766-6830d7e19276)
![Image](https://github.com/user-attachments/assets/5e707015-703f-4db6-857c-f37a809450a1)

**Comment by dreamsyntax** [comment]
@MadGamerHD Assuming you are trying to map over an existing file, its best to take the Right To Render and other attributes from the original .DFF, and import them (including adding the sections) in the exported DFF.

**Comment by MadGamerHD** [comment]
> [@MadGamerHD](https://github.com/MadGamerHD) Assuming you are trying to map over an existing file, its best to take the Right To Render and other attributes from the original .DFF, and import them (including adding the sections) in the exported DFF.

When I export using the Atomic Pipeline set to Custom, enter 0x0120 as the Plugin ID, and set the right to render to 5, it works perfectly. Although it's not explicitly labeled as "Right to Render" in RWAnalyze, it functions correctly in-game and doesn't cause any crashes.

This setup could be used as a preset option for users when exporting custom vehicles, so they don't get confused about what values to use. It would make the export process much more user-friendly, especially for those unfamiliar with RenderWare pipelines.

![Image](https://github.com/user-attachments/assets/9c91b023-e526-4c9e-8004-235020ffc3c9)

**Comment by MadGamerHD** [comment]
Before importing the model into Blender, I used RWAnalyze to replace all "Right to Render" extensions in the .dff file with empty extensions. After that, I imported the cleaned model into Blender, made my edits, and re-exported it.

The final exported .dff works perfectly fine in-game, with no crashes or rendering issues. This proves that "Right to Render" is not required for vehicle models, as long as the rest of the structure (geometry, dummies, materials) is intact.

This method can be really useful for custom vehicle exporting, and it would be great to have an option for this in vehicle export tools—skipping or stripping the Right to Render extension automatically.

previews

![Image](https://github.com/user-attachments/assets/7c4a0bce-5cb2-47f0-b82c-6362ee74bdef)

![Image](https://github.com/user-attachments/assets/969bcd9d-9549-45df-886d-2c8f381c4338)
